src/fastfield/serializer/gcd.rs

```diff
@@ -0,0 +1,176 @@
+use fastdivide::DividerU64;
+use gcd::Gcd;
+
+pub const GCD_DEFAULT: u64 = 1;
+
+pub fn find_gcd(mut numbers: impl Iterator<Item = u64>, base: u64) -> Option<u64> {
+    let mut num1 = 0;
+    let mut num2 = 0;
+    loop {
+        let num = numbers.next()? - base;
+        if num1 == 0 {
+            num1 = num;
+        }
+        if num2 == 0 {
+            num2 = num;
+        }
+        if num1 != 0 && num2 != 0 {
+            break;
+        }
+    }
+    let mut gcd = (num1).gcd(num2);
+    if gcd == 0 {
+        return None;
```

Comment:
So this branch happens if all numbers are zero?

What about we simply start the function by 

```
let mut non_null_numbers = numbers.filter(|n| *n != 0);
```

and remove 12 lines of code?
```

Reply:
yes that's better